### PR TITLE
Add Amazon Comprehend Document Classifier

### PR DESCRIPTION
**Pull request by gopidesupavan**
Adding Amazon Comprehend document classifier. Doc, Operator, Sensor, Trigger, Waiter, Unit Test, System Test.

Manually tested in Breeze with

wait_for_completion=False with a Sensor
deferrable=True.
wait_for_completion=True

For the system test, I used two documents from AWS samples and created multiple copies. Since the classifier requires a minimum of 10 documents for training for each label. I've observed that it takes a maximum of 10 to 15 minutes to train the classifier, given the limited number of labels and documents. This is the minimum setup I was able to get running, so it can be executed in the daily system test suite.

image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/system/providers/amazon/aws/example_comprehend_document_classifier.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    # [END howto_operator_create_document_classifier]
+    create_document_classifier.wait_for_completion = False
+
+    # [START howto_sensor_create_document_classifier]
+    await_create_document_classifier = ComprehendCreateDocumentClassifierCompletedSensor(
+        task_id="await_create_document_classifier", document_classifier_arn=create_document_classifier.output
+    )
+    # [END howto_sensor_create_document_classifier]
+
+    @task
```
Comment (vincbeck) [inline tests/system/providers/amazon/aws/example_comprehend_document_classifier.py]:
```suggestion
    @task(trigger_rule=TriggerRule.ALL_DONE)
```

File: tests/system/providers/amazon/aws/example_comprehend_document_classifier.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    chain(
+        create_document_classifier,
+        await_create_document_classifier,
+        delete_classifier(create_document_classifier.output),
+    )
+
+
+@task
+def copy_data_to_s3(bucket: str, sources: list[dict], prefix: str, number_of_copies=1):
```
Comment (vincbeck) [inline tests/system/providers/amazon/aws/example_comprehend_document_classifier.py:121]: Have you considered using operators such as `HttpToS3Operator` or `LocalFilesystemToS3Operator`?

**Review by vincbeck — COMMENTED** [review]
Just 2 comments but overall fantastic work!!

File: tests/system/providers/amazon/aws/example_comprehend_document_classifier.py (hunk 1 above)
Comment (gopidesupavan) [reply tests/system/providers/amazon/aws/example_comprehend_document_classifier.py]: Updated.

File: tests/system/providers/amazon/aws/example_comprehend_document_classifier.py (hunk 2 above)
Comment (gopidesupavan) [reply tests/system/providers/amazon/aws/example_comprehend_document_classifier.py:121]: These are actually useful here, somehow i forgot. I have updated now, copy data to use HttpToS3Operator. Thank you @vincbeck. 😄

**Comment by gopidesupavan** [comment]
> Just 2 comments but overall fantastic work!!

Thank you 😄

**Review by vincbeck — APPROVED** [review]

**Pull request merged**